File: src/main/java/com/winter/mayawinterfox/command/impl/status/CommandPing.java
package com.winter.mayawinterfox.command.impl.status;

import com.winter.mayawinterfox.checks.PermissionChecks;
import com.winter.mayawinterfox.command.Command;
import com.winter.mayawinterfox.data.Node;
import com.winter.mayawinterfox.util.StatusUtil;
import sx.blah.discord.handle.obj.Permission;

import java.util.Collections;
import java.util.HashSet;

public class CommandPing extends Node<Command> {

	public CommandPing() {
		super(new Command(
				"ping",
				"ping-help",
				PermissionChecks.hasPermission(Permission.SEND_MESSAGES),
				StatusUtil::ping,
				new HashSet<>(Collections.singleton("pong"))
		), Collections.emptyList());
	}
}

File: src/main/java/com/winter/mayawinterfox/command/impl/status/CommandPing.kt
package com.winter.mayawinterfox.command.impl.status

import com.winter.mayawinterfox.command.Command
import com.winter.mayawinterfox.command.CommandPermission
import com.winter.mayawinterfox.util.MessageUtil
import discord4j.core.`object`.util.Permission
import discord4j.core.`object`.util.PermissionSet
import discord4j.core.event.domain.message.MessageCreateEvent
import reactor.core.publisher.Mono

class CommandPing : Command(
		"ping",
		"ping-help",
		Command.Category.STATUS,
		CommandPermission(PermissionSet.of(Permission.SEND_MESSAGES), PermissionSet.of(Permission.SEND_MESSAGES, Permission.ADMINISTRATOR)),
		setOf("pong")
) {
	override fun call(event: MessageCreateEvent): Mono<Void> {
		return event.message.channel
				.flatMap { c -> MessageUtil.sendMessage(c, "pong", event.client.responseTime) }
				.then()
	}
}